<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="yolo" default="true"/>
	<arg name="gazebo" default="false"/>


  <group unless="$(arg yolo)" >
     <include file="$(find object_detection)/launch/find_objects.launch" />
  </group>

	<group unless="$(arg gazebo)">
		<node name="update_robot_in_simulation" pkg="gazebo_simulation_generator" type="update_robot_in_simulation.py" respawn="false" output="screen"/>
	</group>

  <node name="object_detect" pkg="gazebo_simulation_generator" type="object_detect.py" respawn="false" output="screen">
		<param name="~use_yolo" value="$(arg yolo)"/>
	</node>

  <node name="spawn_server" pkg="gazebo_simulation_generator" type="gazebo_spawn_server.py" respawn="false" output="screen">
		<param name = "~use_gazebo" value ="$(arg gazebo)"/>
	</node>

  <node name="spawn_client" pkg="gazebo_simulation_generator" type="gazebo_spawn_client.py" respawn="false" output="screen">
		<param name = "~use_gazebo" value ="$(arg gazebo)"/>
	</node>


</launch>

